python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/perf/partition_mappings_galore_perf_scenario.py

```diff
@@ -0,0 +1,96 @@
+from typing import Optional, Sequence
+
+from dagster import (
+    AssetDep,
+    AssetsDefinition,
+    AutoMaterializePolicy,
+    DailyPartitionsDefinition,
+    Definitions,
+    HourlyPartitionsDefinition,
+    PartitionsDefinition,
+    RunRequest,
+    TimeWindowPartitionMapping,
+    asset,
+)
+from dagster._core.storage.tags import (
+    ASSET_PARTITION_RANGE_END_TAG,
+    ASSET_PARTITION_RANGE_START_TAG,
+)
+
+from .perf_scenario import ActivityHistory, PerfScenario
+
+
+def asset_def(
+    key: str,
+    deps: Sequence[str],
+    partitions_def: Optional[PartitionsDefinition] = None,
+) -> AssetsDefinition:
+    @asset(
+        name=key,
+        partitions_def=partitions_def,
+        deps=deps,
+        auto_materialize_policy=AutoMaterializePolicy.eager(),
+    )
+    def _asset() -> None:
+        ...
+
+    return _asset
+
+
+hourly_one_year = HourlyPartitionsDefinition(start_date="2022-10-18-00:00")
```

Comment:
should these theoretically have end dates to avoid this test getting progressively more expensive as time goes on? there's also some weirdness going on with the fact that we generate the snapshot today, which will fill in every existing partition, but when we run the test tomorrow, the most recent partition will be missing, which may have different implications for perf.

Reply:
Great point. I updated the PR to have the scenario specify an evaluation time, because I think that's a little more realistic than the PartitionsDefinition itself having an end date.